src/content-scanner/ContentScanner.ts

```diff
@@ -88,29 +87,44 @@ export class ContentScanner {
         }
 
         // eslint-disable-next-line no-async-promise-executor
-        const promise = new Promise<boolean>(async resolve => {
+        const promise = new Promise<boolean>(async (resolve, reject) => {
             let response: Response;
 
             if (file) {
                 if (!this.hasKey) {
-                    const k = await fetch(this.scannerUrl + "/_matrix/media_proxy/unstable/public_key")
-                        .then(r => r.json());
-                    this.mcsKey.set_recipient_key(k["public_key"]);
-                    this.hasKey = true;
+                    try {
+                        const k = await fetch(this.scannerUrl + "/_matrix/media_proxy/unstable/public_key")
+                            .then(r => r.json());
+                        this.mcsKey.set_recipient_key(k["public_key"]);
+                        this.hasKey = true;
+                    } catch (err) {
+                        reject(err);
```

Comment:
does catch + reject actually do anything ? Wouldn't throwing in a promise cause the promise to be rejected anyway ?

Reply:
Throwing in a Promise directly yes. But here the errors are raised in other async functions. That would lead to an uncaught in promise error.

Reply:
Doing all in an `async` function works: https://github.com/tchapgouv/tchap-web-v4/pull/132/commits/b5902acfe2bcd30b8c715acb9dd0bef5e7cc12f3